base/loading.jl

```diff
@@ -1306,25 +1307,21 @@ function run_extension_callbacks(pkgid::PkgId)
     # take ownership of extids that depend on this pkgid
     extids = pop!(EXT_DORMITORY, pkgid, nothing)
     extids === nothing && return
+    extids_to_load = Vector{ExtensionId}()
     for extid in extids
-        if extid.ntriggers > 0
-            # indicate pkgid is loaded
-            extid.ntriggers -= 1
-        end
-        if extid.ntriggers < 0
-            # indicate pkgid is loaded
-            extid.ntriggers += 1
-            succeeded = false
-        else
-            succeeded = true
-        end
+        extid.ntriggers -= 1
```

Comment:
This seems like it could result in a priority inversion (since this PR would now document that we care about not doing that), if an extension calls `using` on yet another package, since we will load all of the extensions for that other package immediately, before coming back to here.

Consider if we had these extensions, and B was loaded, then we loaded A.
```
 - Aext (loads C)
 - Aext Bext
 - Aext Bext Cext
```
Observe how, at first, only Aext and Bext were simultaneously eligible to load, so we start on Aext, but loading Aext loaded C, which made Cext eligible to load, and moved Cext in priority ahead of Bext.

Reply:
The fix might be simply to decrement `ntriggers` at the end of the function, after loading all of the extensions, unless `ntriggers==1` (in which case we run it instead)

Reply:
This is ordering by number of triggers, but I don't think that's the priority we care about.

I think the thing we actually care about respecting (and would document) is that we respect the partial order on trigger inclusion (i.e. ext w/ triggers `{A,B}` loads before ext w/ triggers `{A,B,C}`)

IIUC, that priority doesn't get inverted so this is OK?

Reply:
@vtjnash would be good to know if you agree with this, since we might need this for #56368

Reply:
OK I'm pretty convinced we don't care about this inversion.

We only guarantee loading order when loading one extension implies loading another just based on its triggers alone (i.e. _without_ knowledge of its loading / dependency graph), sowe don't have to worry about maintaining ordering over `using` edges.

The only thing we have to worry about (which this solves) is extensions who become simultaneously eligible to load during a single atomic "trigger registration" event